test/unit/smart-commit.test.js

```diff
@@ -253,6 +254,16 @@ describe('Smart commit parsing', () => {
         issueKeys: ['JRA-090']
       })
     })
+
+    it('uppercases issue keys', () => {
+      const text = 'jra-090'
```

Comment:
I've gone back and forth on this, but in general it's always seemed best to ignore lowercases so we eliminate a whole class of false positives. Branch names especially are almost always lower case and have dashes in them. I can't recall any feedback where people requested to be able to lower case issue keys, and all of the entities in the official QA repositories are capitalized: https://github.com/fusion-qa

Reply:
Your point about false positives makes sense and the [Atlassian docs](https://confluence.atlassian.com/bitbucket/use-smart-commits-298979931.html) don't mention supporting lower case keys either.

The reason I added this was because of https://github.com/integrations/jira/issues/188. A user reported that DVCS was able to pull lower case tickets out of branch names:

> When using the DVCS Connector a branch name feature/lief-12155-test-jira was being correctly linked to the ticket LIEF-12155 in JIRA. This doesn't work anymore with the new GitHub for Jira app. I have to use feature/LIEF-12155-test-jira (please note the capitalised project name).
> – https://github.com/integrations/jira/issues/188

I hadn't considered it until now but perhaps DVCS made an exception for branches, allowing lower case issue keys. Do you think that would be a better approach?

In the meantime, I'll try to setup DVCS on my developer Atlassian account to confirm the behaviour is scope to branches only.

Reply:
After playing with DVCS, I found that it also does not support lower case issue keys, except when they appear at the beginning of a branch name.

```bash
feature/ws-1-something-good # doesn't associate to WS-1
ws-1-something-good         # does associate to WS-1
```

Reply:
I agree with not supporting lower-case identifiers. All caps is the de facto standard for JIRA and within the JIRA UX this is how it is always presented and to my knowledge, everybody does it like that. 
Since it makes the solution more resilient to only support uppercase, I suggest doing so until somebody actually asks for lower case support.

Reply:
:+1: That seems reasonable. I'll drop support for lower case identifiers.

I'll open an issue to capture the special case for branches starting with lower case identifiers in the DVCS Connector to explore adding later.